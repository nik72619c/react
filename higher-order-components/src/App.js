import React, { Component } from 'react';

import DummyComponent from './dummyComponent';
import './App.css';

class App extends Component {

  
  handler(){
    console.log('refs', this);   //points to hoc
  }
  render() {
    return (
      <div>
        <DummyComponent/>
        {/* <DummyComponent/> */}
        <button onClick={this.handler.bind(this)}>check ref here</button>
      </div>
    );
  }
}

export default App;
