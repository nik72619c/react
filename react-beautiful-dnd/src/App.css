.listitem{
    padding: 10px ;
    marginBottom: 3px;
    border: 2px solid black;

}