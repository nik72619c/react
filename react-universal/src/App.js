import React from 'react';
import {Switch,Route} from 'react-router';
import Home from './pages/homePageComponent';
class App extends React.Component{
    render(){
        return(
            <Switch>
                <Route path="/home" component={Home}/>
            </Switch>
        )
    }
}

export default App;