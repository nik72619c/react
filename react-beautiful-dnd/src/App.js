import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { DragDropContext,Droppable,Draggable} from 'react-beautiful-dnd';

class App extends Component {

  onDragEnd(){

  }
  render() {
    return (
      <DragDropContext onDragEnd={this.onDragEnd.bind(this)}>
      <div className="App">
      <Droppable droppableId={1}>
   {
     (provided)=>{
       return <div {...provided.droppableProps} innerRef={provided.innerRef}>
           <Draggable draggableId={1} index={1}>
             {
               (provided)=>{
                 return  <div style={{
                  border: '1px solid black',
                  padding: '10px' ,marginBottom: '3px'
                }} {...provided.draggableProps} {...provided.dragHandleProps} innerRef={provided.innerRef}>list item 1</div>
               }
             }
           </Draggable>

            <Draggable draggableId={2} index={2} {...provided.draggableProps} {...provided.dragHandleProps} innerRef={provided.innerRef}>
             {
               (provided)=>{
                 return  <div style={{
                  border: '1px solid black',
                  padding: '10px' ,marginBottom: '3px'
                }} {...provided.draggableProps} {...provided.dragHandleProps} innerRef={provided.innerRef}>list item 2</div>
               }
             }
           </Draggable>

            <Draggable draggableId={3} index={3}>
             {
               (provided)=>{
                 return  <div style={{
                  border: '1px solid black',
                  padding: '10px' ,marginBottom: '3px'
                }} {...provided.draggableProps} {...provided.dragHandleProps} innerRef={provided.innerRef}>list item 3</div>
               }
             }
           </Draggable>

 
      {provided.placeholder}
       </div>
     }
   }
      </Droppable>
      </div>
      </DragDropContext>
    );
  }
}

export default App;
