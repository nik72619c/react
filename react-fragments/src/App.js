import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <>   //react fragment shorter syntax

        <p>this is a p tag</p>
        <p>another p tag without being wrapped</p>
      </>
    );
  }
}

export default App;
