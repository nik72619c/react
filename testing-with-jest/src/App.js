import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(){
    super();
    this.state={
      count:0
    }
  }
  handleClick(){
    this.setState({
      count: this.state.count+1
    });
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            {this.state.count}
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer" onClick={this.handleClick.bind(this)}
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
