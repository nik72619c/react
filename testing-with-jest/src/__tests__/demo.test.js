import React from 'react';
import {render,Simulate,cleanup,fireEvent, getByText} from 'react-testing-library';
import App from '../App';
import TestRenderer from 'react-test-renderer';

test('check if app renders correclty',()=>{
    let component=render(<App/>);
//    let el= component.getByText('Learn React');
//    expect(el).toMatchSnapshot();

fireEvent.click(component.getByText('Learn React'));
let dom=TestRenderer.create(<App/>);
let tree=dom.toJSON();
// let element=tree.children.filter(domElement=>domElement.props.className==='App-link');
expect(dom).toMatchSnapshot();
});

