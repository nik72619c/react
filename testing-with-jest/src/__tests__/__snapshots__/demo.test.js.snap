// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if app renders correclty 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      0
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
